### PR TITLE
updated vendor and client models and routes, removed auth for vendor list

### PR DESCRIPTION
**Pull request by SergeDot**
- changed vendor and client models to contain the contacts and addresses directly instead of a reference to a Schema. Potentially will create a parent class user for them to avoid code duplication. to be discussed
- added property validations
- removed auth requirement for vendor list
- added GET All, PATCH, DELETE routes

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
-  "name": "bb-practicum-team-1-back",
+  "name": "Beauty Finder GG-PRAC-TEAM5",
```
Comment (Pisotski) [inline package.json:2]: I don't know if that matters, yet according to CTD's pattern this should be
gg-practicum-team-5-back. I leave it up to you

File: package.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
   "version": "1.0.0",
-  "description": "Back-End Repo for Team 1 of Baboon/Bald Eagle Practicum",
+  "description": "Back-End Repo for GG Team 5",
```
Comment (Pisotski) [inline package.json:4]: same here
"Back-End Repo for Team 5 of Grackle/Giraffe",

**Review by Pisotski — APPROVED** [review]
All looks good.
Some heavy lifting done with data validation. Thank you.
Check out https://dbdiagram.io/d or any other db visualizer. If you have time, may I ask to please create some visuals. I'm getting lost in what has to be connected to what in models.

File: package.json (hunk 1 above)
Comment (SergeDot) [reply package.json]: Done

File: package.json (hunk 2 above)
Comment (SergeDot) [reply package.json]: Done

**Pull request merged**